services/ansibler/server/loadbalancers.go

```diff
@@ -448,6 +455,51 @@ func setUpNginx(lb *pb.LBcluster, targetedNodepool []*pb.NodePool, directory str
 	return nil
 }
 
+// setUpNodeExporter sets up node-exporter on the LB node
+// return error if not successful, nil otherwise
+func setUpNodeExporter(lb *pb.LBcluster, directory string) error {
+	//create key files for lb nodepools
+	if _, err := os.Stat(directory); os.IsNotExist(err) {
+		if err := os.MkdirAll(directory, os.ModePerm); err != nil {
+			return fmt.Errorf("failed to create directory %s : %w", directory, err)
+		}
+	}
+	if err := utils.CreateKeyFile(lb.ClusterInfo.PrivateKey, directory, fmt.Sprintf("key.%s", privateKeyExt)); err != nil {
+		return fmt.Errorf("failed to create key file for %s : %w", lb.ClusterInfo.Name, err)
+	}
+
+	// generate node-exporter playbook template
+	templateLoader := templateUtils.TemplateLoader{Directory: templateUtils.AnsiblerTemplates}
+	template := templateUtils.Templates{Directory: directory}
+	tpl, err := templateLoader.LoadTemplate(nodeExporterPlaybookTpl)
+	if err != nil {
+		return fmt.Errorf("error while loading %s template for %s : %w", nodeExporterPlaybookTpl, lb.ClusterInfo.Name, err)
+	}
+	err = template.Generate(tpl, nodeExporterPlaybook, LbPlaybookData{Loadbalancer: lb.ClusterInfo.Name})
+	if err != nil {
```

Comment:
ditto `if err := .. ; err != nil{}`

Reply:
done

Reply:
I think you changed it everywhere but on this line :).

Reply:
ah sorry, my bad

Reply:
This time done